scheduler/src/cook/kubernetes/api.clj

```diff
@@ -780,6 +795,9 @@
         {:keys [volumes volume-mounts sandbox-volume-mount-fn]} (make-volumes volumes sandbox-dir)
         {:keys [custom-shell init-container set-container-cpu-limit? sidecar]} (config/kubernetes)
         checkpoint (calculate-effective-checkpointing-config job task-id)
+        job-submit-time (when (:job/submit-time job) ; due to a bug, submit time may not exist for some jobs
```

Comment:
What's the bug?

Reply:
Don't know. This comment was added https://github.com/twosigma/Cook/commit/0208e4701c32ad1980b1e266258bec5f8bfcf7de#diff-7f153e0aaafb627dc91b7170dc98673fad41bdbca07501ff2247f0f2119563dcR469

Reply:
the submit time is at least null for synthetic pods

Reply:
I'm not understanding what I should be looking at in that link

Reply:
Synthetic pods aren't jobs

Reply:
that link is the commit from Dec 27, 2016 when the comment "due to a bug, submit time may not exist for some jobs" was added

synthetic pods are launched and a pod is created. So this check also happens to cover that case

Reply:
Got it. Sorry, GitHub was collapsing that file even though your deep link was to a line in the file.

Is the "bug" the fact that systems can create jobs directly in the Datomic database instead of going through the API?

Either way, should we extract a function for this since it's going to be in two places?

```
(when (:job/submit-time job)
  (.getTime (:job/submit-time job)))
```

Reply:
updated